suites/tipset/messages_application.go

```diff
@@ -4,14 +4,55 @@ import (
 	"context"
 	"testing"
 
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
 	"github.com/filecoin-project/go-address"
+	"github.com/filecoin-project/specs-actors/actors/abi"
 	big_spec "github.com/filecoin-project/specs-actors/actors/abi/big"
 
 	"github.com/filecoin-project/chain-validation/chain"
+	"github.com/filecoin-project/chain-validation/chain/types"
 	"github.com/filecoin-project/chain-validation/drivers"
 	"github.com/filecoin-project/chain-validation/state"
 )
 
+func TestBlockMessageApplication(t *testing.T, factory state.Factories) {
+	builder := drivers.NewBuilder(context.Background(), factory).
+		WithDefaultGasLimit(1_000_000).
+		WithDefaultGasPrice(big_spec.NewInt(1)).
+		WithActorState(drivers.DefaultBuiltinActorsState)
+
+	t.Run("SECP and BLS messages cost different amounts of gas", func(t *testing.T) {
+		td := builder.Build(t)
+		defer td.Complete()
+		tipB := drivers.NewTipSetMessageBuilder(td)
+		blkB := drivers.NewBlockBuilder(td, td.ExeCtx.Miner)
+
+		senderBLS, _ := td.NewAccountActor(address.BLS, big_spec.NewInt(10_000_000))
+		receiverBLS, _ := td.NewAccountActor(address.BLS, big_spec.Zero())
+		senderSECP, _ := td.NewAccountActor(address.SECP256K1, big_spec.NewInt(10_000_000))
+		receiverSECP, _ := td.NewAccountActor(address.SECP256K1, big_spec.Zero())
+		transferAmnt := abi.NewTokenAmount(100)
+
+		results := tipB.WithBlockBuilder(
+			blkB.
+				WithBLSMessageOk(
+					td.MessageProducer.Transfer(receiverBLS, senderBLS, chain.Nonce(0), chain.Value(transferAmnt))).
+				WithSECPMessageOk(
+					td.MessageProducer.Transfer(receiverSECP, senderSECP, chain.Nonce(0), chain.Value(transferAmnt))),
+		).ApplyAndValidate()
+
+		require.Equal(t, 2, len(results.Receipts))
+
+		blsGasUsed := results.Receipts[0].GasUsed
+		secpGasUsed := results.Receipts[1].GasUsed
+		assert.Equal(t, types.GasUnits(246), blsGasUsed)
+		assert.Equal(t, types.GasUnits(268), secpGasUsed)
```

Comment:
I think it would be sufficient for this test to check just that SECP > BLS, to reduce the maintenance burden.